snapm/_progress.py

```
@@ -0,0 +1,620 @@
+        """
+        self.total = 0
+
+    def start(self, total: int):
+        """
+        Start reporting progress on this ``QuietProgress`` object.
+
+        :param total: The total number of expected progress items.
+        :type total: ``int``
+        """
+        if total <= 0:
+            raise ValueError("total must be positive.")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)